src/lib/components/ResultsTable/ResultsTable.js

```diff
@@ -38,7 +39,14 @@ export class ResultsTable extends Component {
         <ResultsTableHeader columns={columns} />
         <ResultsTableBody
           columns={columns}
-          rows={showAllResults ? data : data.slice(0, showMaxRows)}
+          rows={
+            showAllResults
+              ? data
+              : data.slice(
+                  (currentPage - 1) * showMaxRows,
```

Comment:
did you test what happens if ```currentPage * showMaxRows >data.length```

Reply:
From the javascript slice documentation.
![Screenshot from 2021-11-09 13-42-38](https://user-images.githubusercontent.com/25476209/140926301-3baf2569-691d-4ef5-8c66-05f7608dd378.png)

So it should work (tested).

Reply:
👍